### ISSUE TITLE
Border Radius support

### ISSUE THREAD

**Issue by mrousavy** [mrousavy/react-native-blurhash#6]
Currently, no border radius property is supported. You can achieve the same behaviour in react by setting `overflow: hidden` and then applying your borderRadius.

To natively support borderRadius the property group has to be exposed and set on the native views (`UIImageView`, `ReactImageView`).

For Android the official image view provides this borderRadius prop, [see the react native source code](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageManager.java#L171-L190), not sure why that doesn't work in my case since `BlurhashImageView` is a derived class (from `ReactImageView`)

On iOS, this looks really easy, [see this stackoverflow question](https://stackoverflow.com/questions/4314640/setting-corner-radius-on-uiimageview-not-working). Couldn't find the react native's Image component source code, if anyone can help me out here I'd appreciate it.

**Comment by mrousavy** [comment]
When setting `borderRadius` right now, it does nothing on iOS but makes the Image disappear on Android..

**Comment by mrousavy** [comment]
Supported on iOS through https://github.com/mrousavy/react-native-blurhash/commit/3f7d92148fa2df771ca39f904bf5a833cfded6eb, also enables `shouldRasterize` for the `UIImageView`'s layer.

Android still doesn't support it, no idea how to implement that on a layer level. If anyone has an idea, please comment or open a PR with a solution.

**Comment by mrousavy** [comment]
On second thought, maybe it's not a good idea to try and support that natively. It's probably more efficient and less code to just let the containing view handle the clipping (border radius)

**Comment by gwenoleR** [comment]
Hi @mrousavy 
It seems that in Android even if the containing View has the radius the Blurhash doesnt have it.. Or may be I made a mistake but work fine on iOS
![image](https://user-images.githubusercontent.com/10418241/108400332-62129300-721b-11eb-962c-7772bbe3049d.png)
Left: Android
Right: iOS

**Comment by mrousavy** [comment]
@gwenoleR yes, overflow hidden is the key here. I'm still not sure if I want to natively support border radius or just leave it up to the user to wrap it in an overflow hidden view 🤔

**Comment by andreialecu** [comment]
One other interesting problem here is that if you set `borderRadius` to a large value on iOS it begins "inverting/collapsing" the border until it disappears.

This is with borderRadius: 100
<img width="78" alt="Screenshot 2021-05-23 at 16 16 56" src="https://user-images.githubusercontent.com/697707/119261980-4c2db900-bbe2-11eb-8660-f494f434fc7b.png">

Originally I had `borderRadius` set to `Number.MAX_SAFE_INTEGER` and nothing was rendering.

**Comment by mrousavy** [comment]
@andreialecu let me know if https://github.com/mrousavy/react-native-blurhash/pull/119 fixes this for you.
